### PR TITLE
Add enterprise user extension

### PR DESCRIPTION
**Pull request by Captain-P-Goldfish**
Rebuild commit for schema extensions for easier handling and added enterprise user extension

### CODE REVIEW

**Comment by Captain-P-Goldfish** [comment]
@IsuraD , @JKAUSHALYA, @IshikaGodage 
would it be possible that anyone of you would review the pull requests here?

Best regards
Pascal Knüppel

**Comment by Captain-P-Goldfish** [comment]
just by the way. I got a branch that resolves all merge conflicts currently occuring with some of the pull requests

File: modules/charon-core/pom.xml [hunk 1]
```diff
@@ -46,16 +46,22 @@
...
-        </dependency>
         <dependency>
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-api</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-lang3</artifactId>
+        </dependency>
+
```
Comment (sarubi) [inline modules/charon-core/pom.xml]: Remove extra line

File: modules/charon-core/src/main/java/org/wso2/charon3/core/objects/EnterpriseUser.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+/**
+ * author Pascal Knueppel <br>
+ * created at: 18.03.2019 - 16:15 <br>
+ * <br>
+ * this class serves as a wrapper class for user that helps reading the enterprise user attributes
+ */
+public class EnterpriseUser extends ScimAttributeAware {
+
+    /**
+     * a user object that might hold an enterprise extension
```
Comment (sarubi) [inline modules/charon-core/src/main/java/org/wso2/charon3/core/objects/EnterpriseUser.java]: Please follow comment convention in all places, Start with caps and end with a full stop.

File: pom.xml [hunk 3]
```diff
@@ -119,9 +140,40 @@
...
+                <groupId>commons-io</groupId>
+                <artifactId>commons-io</artifactId>
+                <version>${commons.io.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.commons</groupId>
+                <artifactId>commons-lang3</artifactId>
+                <version>${commons.lang3.version}</version>
+            </dependency>
+
```
Comment (sarubi) [inline pom.xml]: Shall we remove the extra lines here?

File: modules/charon-core/src/test/resources/org/wso2/charon3/core/group/create-group-bremen.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+	"schemas": [
+		"urn:ietf:params:scim:schemas:core:2.0:Group"
+	],
+	"externalId": "123456",
+	"displayName": "Bremen"
+}
```
Comment (sarubi) [inline modules/charon-core/src/test/resources/org/wso2/charon3/core/group/create-group-bremen.json]: Shall we keep a new line at EOF

**Comment by DMHP** [comment]
@Captain-P-Goldfish
Can you please address above comments and please note that the travis-ci build is also failing. And we have some conflicts in the PR. Can you please check on those too?

**Comment by GDRDABARERA** [comment]
Merged to the feature branch with https://github.com/wso2/charon/pull/240. Hence closing

**Pull request closed without merging**